src/other/metadata.jl

```diff
@@ -0,0 +1,91 @@
+# Defining behavior for DataFrames metadata
+abstract type AbstractMetaData end
+
+mutable struct MetaData <: AbstractMetaData
+    columndata::Dict{Symbol, Vector}
```

Comment:
`struct` would be enough, right?

Also, `columndata` is a bit of a weird name for this, since it sounds like there are other fields with non-column data. Maybe just `dict`?

Reply:
changed.